### ISSUE TITLE
Post install script for app extensions stopped working

### ISSUE THREAD

**Issue by arctouch-yurireis** [PureLayout/PureLayout#245]
The script described here https://github.com/PureLayout/PureLayout/wiki/App-Extensions used to work perfectly until some recent changes on CocoaPods. On CocoaPods 1.8.0 ~ 1.8.4 the script fails to enter the conditional block, meaning that the `GCC_PREPROCESSOR_DEFINITIONS` are never set. 

While debugging the script, I found out that the `APPLICATION_EXTENSION_API_ONLY` flag value is not set during the execution (it's blank). 

If I revert the CocoaPods version to 1.7.5, the script works fine.

Any ideas?

**Comment by luximetr** [comment]
> The script described here https://github.com/PureLayout/PureLayout/wiki/App-Extensions used to work perfectly until some recent changes on CocoaPods. On CocoaPods 1.8.0 ~ 1.8.4 the script fails to enter the conditional block, meaning that the `GCC_PREPROCESSOR_DEFINITIONS` are never set.
> 
> While debugging the script, I found out that the `APPLICATION_EXTENSION_API_ONLY` flag value is not set during the execution (it's blank).
> 
> If I revert the CocoaPods version to 1.7.5, the script works fine.
> 
> Any ideas?

Also have problem with this script. If found temporary solution till this problem will not be solved. You just need remove checking for this flag APPLICATION_EXTENSION_API_ONLY.

Script will look like this
```
post_install do |installer|
  installer.pods_project.targets.each do |target|
    if target.name.end_with? "PureLayout"
      target.build_configurations.each do |build_configuration|
        build_configuration.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] = ['$(inherited)', 'PURELAYOUT_APP_EXTENSIONS=1']
      end
    end
  end
end
```

**Comment by luximetr** [comment]
Hope this will fix soon.

**Comment by toohotz** [comment]
Will take a look into this over the weekend, thank you for catching this.

**Comment by dathorntonsbgtv** [comment]
Any update on this?  Can confirm it's still an issue with Cocoapods 1.9.3

**Comment by dathorntonsbg** [comment]
Any update on this? Can confirm it's still an issue with Cocoapods 1.9.3

**Comment by pronebird** [comment]
@toohotz Hey, did you figure that out?

**Comment by pronebird** [comment]
Just wanted to point out that the script mentioned in the Wiki is not necessary anymore, as I can clearly see that `APPLICATION_EXTENSION_API_ONLY = YES` even when the script is not used (Xcode 12 & CocoaPods 1.7.5 here).

**Comment by igor-makarov** [comment]
I've looked into this as part of https://github.com/CocoaPods/CocoaPods/issues/10087 and it seems that the desired behavior can be achieved more portably by using availability checks:

```objc
if (@available(iOSApplicationExtension 9.0, *)) {
} else {
}
```

In any case, the PureLayout app extension integration has stopped working due its reliance on an implementation detail in CocoaPods that has changed. Please refer to the issue referenced above for more details.

**Comment by toohotz** [comment]
@igor-makarov thanks for pointing out the change in Cocoapods 1.9.3.  At the time when I looked into it earlier this year, the issue still existed and I wasn't able during that time to find a viable solution.  

I have not delved into the Xcode 12 related app extension issue but if I'm correct you're stating even with the **@available** check, the reliance on the implementation detail is still causing an issue with the PL app extension integration, correct?

**Comment by igor-makarov** [comment]
I'm saying you can do the RTL check without relying on a compiler definition like `PURELAYOUT_APP_EXTENSIONS=1` but on a runtime check instead.

**Comment by pronebird** [comment]
@igor-makarov the compiler still throws an error even when wrapping the call in @available(iOSApplicationExtension 9.0, *):

```
'sharedApplication' is unavailable: not available on iOS (App Extension) - Use view controller based solutions where appropriate instead.
```

**Comment by igor-makarov** [comment]
@pronebird the order should be reversed:

```
BOOL isRightToLeftLayout;
if (@available(iOSApplicationExtension 9.0, *)) {
        isRightToLeftLayout = [[UIApplication sharedApplication] userInterfaceLayoutDirection] == UIUserInterfaceLayoutDirectionRightToLeft;
NSLocaleLanguageDirectionRightToLeft;
} else {
        // App Extensions may not access -[UIApplication sharedApplication]; fall back to checking the bundle's preferred localization character direction
        isRightToLeftLayout = [NSLocale characterDirectionForLanguage:[[NSBundle mainBundle] preferredLocalizations][0]] == }
```

**Comment by pronebird** [comment]
@igor-makarov I don't think so, but even reversing the order does not solve that. 
<img width="1508" alt="Screen Shot 2020-11-16 at 09 22 44" src="https://user-images.githubusercontent.com/704044/99230082-2799f400-27ef-11eb-91d8-a64acbf4bf05.png">

**Comment by igor-makarov** [comment]
@pronebird you're right, my mistake about the syntax. 

My next suggestion would be to use the extension-safe API and stop using the `UIApplication`-based API. From what I understand, this approach is valid for iOS apps, extensions and everywhere else.

**Comment by toohotz** [comment]
Fixed in #252, thanks everyone. 🙂
